ethsigner/signer/azure/src/test/java/tech/pegasys/ethsigner/signer/azure/AzureKeyVaultAuthenticatorTest.java

```diff
@@ -168,4 +165,51 @@ public void nullClientAndOrSecretAreHandledCleanly() {
     assertThatThrownBy(() -> authenticator.getAuthenticatedClient(null, null))
         .isInstanceOf(IllegalArgumentException.class);
   }
+
+  private static class AzureConfigBuilder {
+
+    private String keyvaultName;
+    private String keyName;
+    private String keyVersion;
+    private String clientId;
+    private String clientSecret;
+
+    public AzureConfigBuilder withKeyvaultName(String keyvaultName) {
+      this.keyvaultName = keyvaultName;
+      return this;
+    }
+
+    public AzureConfigBuilder withKeyName(String keyName) {
+      this.keyName = keyName;
+      return this;
+    }
+
+    public AzureConfigBuilder withKeyVersion(String keyVersion) {
+      this.keyVersion = keyVersion;
+      return this;
+    }
+
+    public AzureConfigBuilder withClientId(String clientId) {
+      this.clientId = clientId;
+      return this;
+    }
+
+    public AzureConfigBuilder withClientSecret(String clientSecret) {
+      this.clientSecret = clientSecret;
+      return this;
+    }
+
+    public AzureConfig build() {
+      return new AzureConfig(keyvaultName, keyName, keyVersion, clientId, clientSecret);
+    }
+  }
+
+  private AzureConfigBuilder createValidConfigBuilder() {
```

Comment:
nit: the params are in a different order than in AzureConfig

Reply:
done